mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -141,6 +135,48 @@ def __init__(self, name=None, bus=None, use_settings=True):
         self.bind(bus)
         #: Mycroft global configuration. (dict)
         self.config_core = Configuration.get()
+
+        self.settings = None
+
+        #: To not break existing setups,
+        #: save to skill directory if the file exists already
+        self.settings_write_path = Path(self.root_dir)
+
+        #: Otherwise save to XDG_CONFIG_DIR
+        if not self.settings_write_path.joinpath('settings.json').exists():
+            self.settings_write_path = Path(BaseDirectory.save_config_path(
+                'mycroft', 'skills', self.name))
```

Comment:
possibly this should be the dirname and not self.name since self.name is settable by the skill and may be changed over time while the skill directory name will always be the same unless a user manually moves the skill (or the package moves the skill)